nvflare/apis/utils/decomposers/flare_decomposers.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Decomposers for objects used by NVFlare itself
+
+This module contains all the decomposers used to run NVFlare.
+The decomposers are registered at server/client startup.
+
+"""
+import os
+from argparse import Namespace
+from typing import Any
+
+from nvflare.apis.client import Client
+from nvflare.apis.dxo import DXO
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.fl_snapshot import RunSnapshot
+from nvflare.apis.shareable import Shareable
+from nvflare.apis.signal import Signal
+from nvflare.apis.workspace import Workspace
+from nvflare.fuel.utils import fobs
+from nvflare.fuel.utils.fobs.decomposer import Decomposer
+
+
+class ShareableDecomposer(Decomposer):
+    @staticmethod
+    def supported_type():
+        return Shareable
+
+    def decompose(self, target: Shareable) -> Any:
+        return target.copy()
+
+    def recompose(self, data: Any) -> Shareable:
+        obj = Shareable()
+        for k, v in data.items():
+            obj[k] = v
+        return obj
```

Comment:
Can you add a test on a Shareable that has a header of Shareable?

usage refer:

https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/private/fed/server/fed_server.py#L597-L612

Reply:
Added